zerver/lib/actions.py

```diff
@@ -1049,8 +1056,9 @@ def check_message(sender, client, message_type_name, message_to,
         message.pub_date = timezone.now()
     message.sending_client = client
 
-    if not message.maybe_render_content(realm.domain):
-        raise JsonableError(_("Unable to render message"))
+    # We are going to assume that any message can be rendered, and if it fails,
+    # it will have to fail later.
+    assert message.rendered_content is None
```

Comment:
So there never was true error handling here, because the call stack under `maybe_render_content()` doesn't have any kind of True/False return protocol.  Basically, we expect all messages to parse, and if they don't, it's likely that an exception will be thrown due to some bug, so that error handling would not have caught anything.

We don't have any tests that exercised line 1053, and I'm guessing it's basically for the reasons I described above.  I try to explain this in the commit message, but I'm happy to reword that or elaborate some more.

The other problem with rendering at this point is that we don't have enough to render, if we are going to pass in the list of users attached to the message, which is the end game of this PR.  Computing the list of users who will get a message is a pretty expensive calculation that happens later in the call.